### PR TITLE
ba-96 Implement method to reset the checkbox of the search bar in the socobo-element-search

### PR DESCRIPTION
**Pull request by lyio**
ba-96 Implement method to reset the checkbox of the search bar in the socobo-element-search

Add resetQuery method to the public API of the search element
Reset the search string in this method
Use double data-binding to reflect changes to the input

### CODE REVIEW

**Comment by flashback2k14** [comment]
okay. but the main idea for this issue was to create a function for unchecking the checkbox.

File: app/elements/socobo-element-search/socobo-element-search.html [hunk 1]
```diff
@@ -171,9 +180,9 @@
           if (this.$.txtSearchQuery.value.length === 0) {
             this._toggleSearchElements();
           } else {
-            this.$.txtSearchQuery.value = "";
-            this.fire("changed-search-query", { searchQuery: null });
+            this.searchStr = "";
           }
+          this.fire("changed-search-query", { searchQuery: this.searchStr });
```
Comment (flashback2k14) [inline app/elements/socobo-element-search/socobo-element-search.html]: you can call searchQueryChanged()

Comment (lyio) [inline app/elements/socobo-element-search/socobo-element-search.html]: Yes, I just moved that line.

File: app/elements/socobo-element-search/socobo-element-search.html [hunk 2]
```diff
@@ -163,6 +163,17 @@
...
+
+        /**
+         * Method for resetting the search query </br>
+         * @method resetQuery
+         */
+        resetQuery: function() {
+          this.searchStr = "";
+          this.searchQueryChanged();
+          this.$.chbCheckAllItems.checked = false;
+          this.fire("uncheck-all-items");
```
Comment (lyio) [inline app/elements/socobo-element-search/socobo-element-search.html:175]: If there is no further application for this method than after deleting items, the checked items are gone anyway and firing this event serves no purpose anymore.

But maybe firing this event makes deselecting stuff easier, preventing accidentally leaving items checked.

**Comment by lyio** [comment]
@flashback2k14 Please reference ba-84 when doing this and maybe sneak in one or two tests for the recipe component.

**Comment by flashback2k14** [comment]
"I have changed the removeAllChecked to call the new method of lyio and removed the code to update the counter which handles the state of the FAB.
Now it works and I think it's good solution because we don't need to handle the counter for the FAB button (view code) in the handler method for deleting all items.

Is it ok to include these changes directly into this pull request. Because I don't want to create a new one for this."

This i haven't written. Maybe Patrick.

**Comment by patrick-hebner** [comment]
Yes I wrote this. Sorry I missed to create a separate comment for this accidentally. 
@lyio I will refer to ba-84 and attempt to add the tests

**Comment by hetec** [comment]
Yes I wrote this. Sorry I missed to create a separate comment for this accidently

**Comment by patrick-hebner** [comment]
Sorry my smartphone browser had become frozen but the clicks were also performed :(

**Comment by flashback2k14** [comment]
I think close is The right decision. We only need The Tests from This PR.

**Comment by patrick-hebner** [comment]
Ok, but I need to know if we use this new method or another one which only resets the checkbox because this also causes some adaptations in the recipe element

**Comment by flashback2k14** [comment]
Okay, one simple question. What need we / you? The issue say a method for reset The checkbox like this

``` javascript
resetCheckbox: function(){
  this.$.chbCheckAllItems.checked = false;
}
```

Not more, not less!

**Comment by patrick-hebner** [comment]
Yes, that would be enough. But I meant it wouldn't be a problem to use the method thomas has created. It doesn't matter which I use but a decision would be nice.

**Comment by lyio** [comment]
That may be enough, but resetting the search query makes sense as well in this method. Otherwise you delete stuff and your list will be empty.

**Comment by flashback2k14** [comment]
i have implemented the resetCheckbox() function. i think we can now merge the PR.

**Comment by patrick-hebner** [comment]
Ok, both ways work well for resetting the checkbox. But thomas is right, if I only remove searched items the search bar will not be reseted with bastis approach. So the list would be empty and the user would have to close the search bar to see the remaining items again. With the approach of thomas the list of items would be restored automatically.

Therefore it's a question of what we want. I would prefer restoring the list of items after deletion

**Comment by patrick-hebner** [comment]
Since both methods are part of the pull request I would suggest to keep both and merge it now

**Comment by flashback2k14** [comment]
do it! :-)

**Pull request merged**